src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/repository/BookingSearchRepository.kt

```diff
@@ -133,7 +149,33 @@ class BookingSearchRepository(private val namedParameterJdbcTemplate: NamedParam
       },
     )
 
-    return result ?: emptyList()
+    return PageImpl(
+      result ?: emptyList(),
+      (if (isNull(pageable)) Pageable.unpaged() else pageable)!!,
+      count()?.toLong() ?: 0,
+    )
+  }
+
+  private fun buildOrderByAndOffsetClause(pageable: Pageable?): String {
+    if (pageable != null) {
+      val sortOrder = pageable.sort.first()
+      val sortingBy = when (sortOrder.property) {
+        bookingCreatedAt.value -> KEY_BOOKING_CREATED_COLUMN_NAME
+        bookingEndDate.value -> KEY_BOOKING_END_DATE_COLUMN_NAME
+        bookingStartDate.value -> KEY_BOOKING_START_DATE_COLUMN_NAME
+        personCrn.value -> KEY_BOOKING_CRN_COLUMN_NAME
+        else -> KEY_BOOKING_CREATED_COLUMN_NAME
+      }
+      return StringBuilder()
+        .append(" ORDER BY $sortingBy ${sortOrder.direction.name} ")
+        .append(" LIMIT ${pageable.pageSize} ")
+        .append(" OFFSET ${pageable.offset} ").toString()
+    }
+    return ""
+  }
+
+  fun count(): Int? {
+    return namedParameterJdbcTemplate.jdbcTemplate.queryForObject("SELECT count(*) FROM bookings", Int::class.java)
```

Comment:
I have changed the custom query to JPA implementation, so we dont have the number of record count() issue.
Since JPA uses the bookingsearch query with where clause of 'serviceName' , we only retrieve or count specific type of booking. 

I have added integration test to prove that we are only getting count of booking based on serviceName.